stko/optimizers/xtb.py

```diff
@@ -0,0 +1,509 @@
+            check that the structure is completely optimized.
+            ``False`` will drastically speed up the calculation but
+            potentially provide incomplete optimizations and forces
+            :attr:`max_runs` to be ``1``.
+
+        num_cores : :class:`int`, optional
+            The number of cores xTB should use.
+
+        electronic_temperature : :class:`int`, optional
+            Electronic temperature in Kelvin.
+
+        solvent : :class:`str`, optional
+            Solvent to use in GBSA implicit solvation method.
+            For details see
+            https://xtb-docs.readthedocs.io/en/latest/gbsa.html.
+
+        solvent_grid : :class:`str`, optional
+            Grid level to use in SASA calculations for GBSA implicit
+            solvent.
+            Can be one of ``'normal'``, ``'tight'``, ``'verytight'``
+            or ``'extreme'``.
+            For details see
+            https://xtb-docs.readthedocs.io/en/latest/gbsa.html.
+
+        charge : :class:`int`, optional
+            Formal molecular charge.
+
+        num_unpaired_electrons : :class:`int`, optional
+            Number of unpaired electrons.
+
+        unlimited_memory : :class: `bool`, optional
+            If ``True`` :meth:`optimize` will be run without
+            constraints on the stack size. If memory issues are
+            encountered, this should be ``True``, however this may
+            raise issues on clusters.
+
+        """
+
+        if solvent is not None:
+            solvent = solvent.lower()
+            if gfn_version == 0:
+                raise XTBInvalidSolventError(
+                    f'No solvent valid for version',
+                    f' {gfn_version!r}.'
+                )
+            if not is_valid_xtb_solvent(gfn_version, solvent):
+                raise XTBInvalidSolventError(
+                    f'Solvent {solvent!r} is invalid for ',
+                    f'version {gfn_version!r}.'
+                )
+
+        if not calculate_hessian and max_runs != 1:
+            max_runs = 1
+            logger.warning(
+                'Requested that hessian calculation was skipped '
+                'but the number of optimizations requested was '
+                'greater than 1. The number of optimizations has been '
+                'set to 1.'
+            )
+
+        self._xtb_path = xtb_path
+        self._gfn_version = str(gfn_version)
+        self._output_dir = output_dir
+        self._opt_level = opt_level
+        self._max_runs = max_runs
+        self._calculate_hessian = calculate_hessian
+        self._num_cores = str(num_cores)
+        self._electronic_temperature = str(electronic_temperature)
+        self._solvent = solvent
+        self._solvent_grid = solvent_grid
+        self._charge = str(charge)
+        self._num_unpaired_electrons = str(num_unpaired_electrons)
+        self._unlimited_memory = unlimited_memory
+        self.incomplete = set()
```

Comment:
If this is some kind of cache, would it better just to create a generic cache in the parent class later on?

Reply:
It is not a cache per se. It does a different job.

Reply:
Can be replaced later on, although it does do a very different role